mlkem/indcpa.c

```diff
@@ -166,41 +168,85 @@ static unsigned int rej_uniform(int16_t *r,
 // Not static for benchmarking
 void gen_matrix(polyvec *a, const uint8_t seed[KYBER_SYMBYTES], int transposed)
```

Comment:
I do not quite understand why this function looks so different from https://github.com/pq-crystals/kyber/blob/main/avx2/indcpa.c? 
Shouldn't it be essentially the same (just ripping out the AVX2 instrinsics)?

I guess this is mostly because this part of the code handles arbitrary `KECCAK_WAY`. Cannot comment on if it results in the same calls to keccakx4 and keccakx1 as the AVX2 version? 

For which values for `KECCAK_WAY` should this work and for which did you test it?

Reply:
The `KYBER_SYMBYTES` is `32 bytes`. So the function signature in `avx2` hardcoded the number. 
I don't like to have `#ifdef KYBER_K` to handle different value of `KYBER_K`, so our function has 2 for loops to handle all the cases. I tested with arbitrary value of `KECCAK_WAY = 1...8`, it always work. 
The compiler will optimize the bound of each for loop, so the overhead is very very mall.

Reply:
I analyze `getnoise_eta1` and `getnoise_eta2` below, it's seem like we can maximize the performance by having more than one KECCAK_WAY.

Reply:
After careful thoughts, I think it's not worth it to introduce additional KECCAK_WAY.